pkg/receive/handler.go

```diff
@@ -890,7 +890,10 @@ func (h *Handler) distributeTimeseriesToReplicas(
 		}
 	}
 
-	return localWrites, remoteWrites, nil
+	maxBufferedResponses += len(remoteWrites)
+	maxBufferedResponses += len(localTenantsSeen)
```

Comment:
should `maxBufferedResponses` be maximum of `len(remoteWrites)` and `len(localTenantsSeen)`? Not fully understand why lengths are summed here...

Reply:
No, the Receiver has a router+ingester mode (2 in 1) so it needs to be a sum.